tequilapi/endpoints/connection.go

```diff
@@ -287,13 +292,25 @@ type proposalRepository interface {
 }
 
 // AddRoutesForConnection adds connections routes to given router
-func AddRoutesForConnection(router *httprouter.Router, manager connection.Manager,
-	stateProvider stateProvider, proposalRepository proposalRepository, identityRegistry identityRegistry, publisher eventbus.Publisher, addressProvider addressProvider) {
+func AddRoutesForConnection(
+	manager connection.Manager,
+	stateProvider stateProvider,
+	proposalRepository proposalRepository,
+	identityRegistry identityRegistry,
+	publisher eventbus.Publisher,
+	addressProvider addressProvider,
+) func(*gin.Engine) error {
 	connectionEndpoint := NewConnectionEndpoint(manager, stateProvider, proposalRepository, identityRegistry, publisher, addressProvider)
-	router.GET("/connection", connectionEndpoint.Status)
-	router.PUT("/connection", connectionEndpoint.Create)
-	router.DELETE("/connection", connectionEndpoint.Kill)
-	router.GET("/connection/statistics", connectionEndpoint.GetStatistics)
+	return func(e *gin.Engine) error {
+		connGroup := e.Group("/connection")
```

Comment:
I would probably go the other way around this:
* Create an empty group `connGroup := e.Group("")`
* Add each handler like `connGroup.GET("/connection", connectionEndpoint.Status)`

Would be a bit easier to read the actions endpoints as they're next to the func call, but maybe thats just me.